packages/block-library/src/template-part/edit/index.js

```diff
@@ -105,6 +103,14 @@ export default function TemplatePartEdit( {
 	const isEntityAvailable = ! isPlaceholder && ! isMissing && isResolved;
 	const TagName = tagName || areaObject.tagName;
 
+	// The `isSelected` check ensures the `BlockSettingsMenuControls` fill
+	// doesn't render multiple times. The block controls has similar internal check.
```

Comment:
For reference - https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-controls/hook.js